screens/GameModal.tsx

```diff
@@ -0,0 +1,113 @@
+import { Button, Modal, StatusBar, Text, View, SafeAreaView } from 'react-native';
+import { Game } from '../types/Game';
+import { useEffect, useState } from 'react';
+import LobbyScreen from './LobbyScreen';
+import subscribeToGameChanges from '../services/subscribeToGameState';
+import { StyleSheet } from 'react-native';
+import { Team } from '../types/Team';
+import QuestionScreen from './QuestionScreen';
+import LeaderboardScreen from './LeaderboardScreen';
+import FinalLeaderboardScreen from './FinalLeaderboardScreen';
+
+interface GameModalProps {
+    visible: boolean;
+    onClose: () => void;
+    game: Game;
+}
+
+const GameModal: React.FC<GameModalProps> = ({ visible, onClose, game }) => {
+    const [team, setTeam] = useState<Team>();
+    const [gameState, setGameState] = useState<string>(game.gameState);
+    const [currentQuestion, setCurrentQuestion] = useState(game.currentQuestion);
+    const [playerScore, setPlayerScore] = useState(0);
+
+    useEffect(() => {
```

Comment:
subscribes to changes in gamestate and currentquestion in database, will probably need to do the same for score changes but unsure rn